desktop_version/src/FileSystemUtils.cpp

```diff
@@ -144,6 +148,28 @@ char *FILESYSTEM_getUserLevelDirectory()
 void FILESYSTEM_loadFileToMemory(const char *name, unsigned char **mem,
                                  size_t *len, bool addnull)
 {
+        if (strcmp(name, "levels/special/stdin.vvvvvv") == 0) {
+            // this isn't *technically* necessary when piping directly from a file, but checking for that is annoying
+            static std::vector<char> STDIN_BUFFER;
+            static bool STDIN_LOADED = false;
+            if (!STDIN_LOADED) {
+                std::istreambuf_iterator<char> begin(std::cin), end;
+                STDIN_BUFFER.assign(begin, end);
+                STDIN_BUFFER.push_back(0); // there's no observable change in behavior if addnull is always true, but not vice versa
+                STDIN_LOADED = true;
+            }
+
+            auto length = STDIN_BUFFER.size() - 1;
```

Comment:
This can be `size_t`, other than that everything looks good now! I'd still be interested in finding a way to write directly to *mem without the stdin buffer, but since it's static memory I'm not too heartbroken about it.

Reply:
Alright, fixed. I also fixed the indentation here as well.